src/gc.h

```diff
@@ -145,12 +145,14 @@ JL_EXTENSION typedef struct _bigval_t {
     // must be 64-byte aligned here, in 32 & 64 bit modes
 } bigval_t;
 
-// data structure for tracking malloc'd arrays and genericmemory.
-
-typedef struct _mallocarray_t {
-    jl_value_t *a;
-    struct _mallocarray_t *next;
-} mallocarray_t;
+// data structure for tracking malloc'd genericmemory.
+typedef struct _mallocmemory_t {
+    union {
+        uint8_t aligned:1;
```

Comment:
I thought this is UB, since there is an undefined amount of padding the compiler can overwrite if it feels like it

Reply:
Possibly.

In any case, after re-reading this code I realized that using a C Union when both fields need to be active at the same time is kind of an awkward choice.

Will refactor this to simply add a comment about the pointer tagging in the lowest bit.